source/lac/cuda_vector.cu

```diff
@@ -688,18 +688,15 @@ namespace LinearAlgebra
       out << std::endl;
 
       // Copy the vector to the host
-      Number *    cpu_val    = new Number[n_elements];
-      cudaError_t error_code = cudaMemcpy(cpu_val,
+      std::vector<Number> cpu_val(n_elements);
+      cudaError_t error_code = cudaMemcpy(cpu_val.data(),
                                           val,
                                           n_elements * sizeof(Number),
-                                          cudaMemcpyHostToDevice);
+                                          cudaMemcpyDeviceToHost);
```

Comment:
You could use `Utilities::CUDA::copy_to_host(val, cpu_val);`  instead. The function didn't exist at the time and that's why there are silly mistakes like this one.